activerecord/lib/active_record/query_cache.rb

```diff
@@ -3,6 +3,7 @@
 module ActiveRecord
   # = Active Record Query Cache
   class QueryCache
+    # ActiveRecord::Base extends this module, so these methods are available in models.
```

Comment:
I wish there was a way to see what classes are extended in the API docs, so we don't have to call this out each time. And for a way to see methods on the receiving class as if they were defined directly.

Reply:
@zzak 100%, a should be able to discover the API of `ActiveRecord::Base` just by browsing.

Furthermore, the module should not be listed in the docs. The way the code is organized does not belong to the public contract. The method could be defined directly, could be injected via a module, ..., that aspect does not belong to the public interface and we should be able to refactor these modules at will.

Reply:
@zzak an idea would be to have a new directive, something like

```ruby
class C
  include M # :doc:inject
end
```

which would also nodoc `M`, perhaps.

For simple use cases, that would work.

It would not suffice for the use case in which the implementation is spread across multiple ancestors, like what happens with `ActiveRecrod::Base#save`.

Reply:
> is spread across multiple ancestors

Yeah, and also if we show every included method the base objects API docs could get unruly.

One hacky way to do it is nodoc the module then do an explicit `:doc: whatever-method` on the target class, but you lose on the source reference (toggle to "show source").

It's that or we make it easier to to find the extending modules, but sounds like you're rather keep those private?

For example, we already show a list of the included (inherited) modules, it should be similar to get a list of extended modules.

Otherwise yeah, we have to introduce a new directive to RDoc.

Reply:
> but sounds like you're rather keep those private?

If `A::B::C` is a module that is published, renaming it, deleting it, etc. is a breaking change. And that is unnecessary, because in the use cases I have in mind (like this one), ancestor chains are 100% internal. The only thing the user has to be told is that the method exists in the containing, user-facing object.

Reply:
Let me show a concrete example in Zeitwerk.

In order to configure your root directories, you invoke `push_dir`:

```ruby
loader.push_dir(...)
```

Where is `push_dir` implemented? That is not a contract we have to stablish, we only need to define that loaders respond to that method, and it is my job to respect that contract through releases. (It is indeed defined in an internal mixin right now.)

Zeitwerk is not documented via RDoc, it is all in the README, where I have full control. But you see what I mean.